langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java

```diff
@@ -36,13 +38,18 @@
 
     public static BTypedesc getElementType(Object td) {
         BTypedesc bTypedesc = (BTypedesc) td;
-        Type type = bTypedesc.getDescribingType();
+        return getTypeDescValue(bTypedesc.getDescribingType());
+    }
+
+    private static BTypedesc getTypeDescValue(Type type) {
         if (type.getTag() == TypeTags.ARRAY_TAG) {
             return ValueCreator.createTypedescValue(((ArrayType) type).getElementType());
         } else if (type.getTag() == TypeTags.STREAM_TAG) {
             return ValueCreator.createTypedescValue(((StreamType) type).getConstrainedType());
         } else if (type.getTag() == TypeTags.TABLE_TAG) {
             return ValueCreator.createTypedescValue(((TableType) type).getConstrainedType());
+        } else if (type.getTag() == TypeTags.FINITE_TYPE_TAG && ((FiniteType) type).getValueSpace().size() == 1) {
+            return getTypeDescValue(((BValue) (((FiniteType) type).getValueSpace().iterator().next())).getType());
         }
 
         return ValueCreator.createTypedescValue(PredefinedTypes.TYPE_NULL);
```

Comment:
Ideally we shouldn't reach here, right? If so, shouldn't we throw an exception, rather than returning null because the actual issue wouldn't be uncovered until we operate on whatever the construct that was created using this?

Reply:
Requested changes in https://github.com/ballerina-platform/ballerina-lang/pull/32257